### PR TITLE
Terabee

### PR DESCRIPTION
**Pull request by J2thatsme**
Terabee people counting sensors

### CODE REVIEW

File: types/terabee/peopleCountingXLoRa/uplink.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    lifecycle.tpcStopped = !!Number(bits.substr(71, 1));
+
+    emit("sample", { data: sample, topic: "default" });
+    emit("sample", { data: lifecycle, topic: "lifecycle" });
+
+    // HTTP
+  } else {
+    sample.fw = data.count_in;
+    sample.bw = data.count_out;
+    emit("sample", { data: sample, topic: "default" });
```
Comment (mariusgiger) [inline types/terabee/peopleCountingXLoRa/uplink.js:26]: is lifecycle not sent in the HTTP case?

File: types/terabee/peopleCountingXLoRa/meta.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "People counting XL LoRa",
```
Comment (mariusgiger) [inline types/terabee/peopleCountingXLoRa/meta.json:2]: only X?

File: types/terabee/peopleCountingXLoRa/meta.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  "manufacturer": "Terabee",
+  "url": "https://www.terabee.com/shop/people-counting/terabee-people-counting-l/",
+  "description": "Improve facility efficiency with accurate people counting data",
+  "author": "Akenza AG",
+  "firmwareVersion": "V1.0.0",
+  "loraDeviceClass": "A",
+  "availableSensors": ["People Counting"],
+  "outputTopics": ["default", "lifecycle"],
+  "encoding": "JSON",
+  "connectivity": "LORA/HTTP"
```
Comment (mariusgiger) [inline types/terabee/peopleCountingXLoRa/meta.json:13]: this is currently not supported, use LORA

File: types/terabee/peopleCountingXLoRa/lifecycle.schema.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    "multiDevIssue": {
+      "title": "Multi device issue",
+      "type": "boolean",
+      "desciption": "The multi-device is not set up properly or there is an issue with the coordinator or one of the subordinate devices.",
+      "hideFromKpis": true
+    },
+    "tpcStuck": {
+      "title": "TPC stuck",
+      "type": "boolean",
+      "desciption": "The counting software is started but nor responding",
```
Comment (mariusgiger) [inline types/terabee/peopleCountingXLoRa/lifecycle.schema.json:22]: add . at the end to conform with the others

File: types/terabee/peopleCountingXL/meta.json [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  "manufacturer": "Terabee",
+  "url": "https://www.terabee.com/shop/people-counting/terabee-people-counting-xl/",
+  "description": "Compact, small and discrete people counter device optimized for single doorways and narrow passages",
+  "author": "Akenza AG",
+  "firmwareVersion": "V1.0.0",
+  "loraDeviceClass": "A",
+  "availableSensors": ["People Counting"],
+  "outputTopics": ["default"],
+  "encoding": "JSON",
+  "connectivity": "HTTPS/MQTT"
```
Comment (mariusgiger) [inline types/terabee/peopleCountingXL/meta.json:13]: use one or the other

File: types/terabee/peopleCountingXL/default.schema.json [hunk 6]
```diff
@@ -0,0 +1,22 @@
+{
+  "$id": "https://akenza.io/terabee/peopleCountingXL/default.schema.json",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Default",
+  "type": "object",
+  "properties": {
+    "fw": {
+      "title": "Forward",
+      "description": "Absolute count of people moving forward",
```
Comment (mariusgiger) [inline types/terabee/peopleCountingXL/default.schema.json:9]: align the descriptions between the different types

File: types/terabee/peopleCountingM/meta.json [hunk 7]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "People counting M",
```
Comment (mariusgiger) [inline types/terabee/peopleCountingM/meta.json:2]: is there also a M LoRa?

**Review by mariusgiger — COMMENTED** [review]
a few minor comments

File: types/terabee/peopleCountingXLoRa/uplink.js (hunk 1 above)
Comment (J2thatsme) [reply types/terabee/peopleCountingXLoRa/uplink.js:26]: no

File: types/terabee/peopleCountingM/meta.json (hunk 7 above)
Comment (J2thatsme) [reply types/terabee/peopleCountingM/meta.json:2]: No

**Review by mariusgiger — APPROVED** [review]

**Review by simonrieser — APPROVED** [review]

**Pull request merged**